src/idl_gen_cpp.cpp

```diff
@@ -814,19 +852,62 @@ class CppGenerator : public BaseGenerator {
     if (!vs.empty()) {
       code_ += "  static const int32_t values[] = { {{VALUES}} };";
     }
-    auto has_names =
-        num_fields && parser_.opts.mini_reflect == IDLOptions::kTypesAndNames;
+    bool has_names = num_fields && parser_.opts.reflect_names;
     if (has_names) {
       code_ += "  static const char *names[] = {";
       code_ += "    {{NAMES}}";
       code_ += "  };";
     }
+    bool has_attributes = parser_.opts.reflect_attrs && type_attrs.count;
+    if (has_attributes) {
+      code_ += "  static const flatbuffers::AttributeList<{{ATTR_COUNT}}>"
+               " attrs = {{ATTR_LIST}};";
+    }
+    static const std::string kAttrListCast =
+        "reinterpret_cast<flatbuffers::RawAttributeList>";
+    has_field_attributes &= parser_.opts.reflect_attrs;
+    if (has_field_attributes) {
+      std::string field_attr_decl;
+      std::map<std::string, std::string> attr_interning;
+      // Allow reusing our Table's attribute list for its attributes.
+      attr_interning[type_attrs.initializer] = "attrs";
+      code_ += "# define DECL_FA_LIST(id, size) "
```

Comment:
I'd recommend against macros in the generated code, as this doesn't help the compiler nor the reader, if any.